### PR TITLE
[ML] add GET _cat/ml/datafeeds

### LINKED ISSUE #51411: Create GET _cat/ml/datafeeds API
See https://github.com/elastic/elasticsearch/pull/51364 for more info on how to do this.

This API should return basic statistics and settings that affect those statistics. 

I think the `GetDatafeedsStatsAction` is a good place to start.

### PR DESCRIPTION
**Pull request by benwtrent**
This adds GET _cat/ml/datafeeds && _cat/ml/datafeeds/{datafeed_id}

Example calls
```
# GET _cat/ml/datafeeds?v
#! Deprecation: [interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future.
id                             state bucket.count search.count
datafeed-low_request_rate    stopped 1457         3
datafeed-response_code_rates stopped 1460         18
datafeed-url_scanning        stopped 1460         18

# GET _cat/ml/datafeeds?h=*&v
#! Deprecation: [interval] on [date_histogram] is deprecated, use [fixed_interval] or [calendar_interval] in the future.
id                             state assignment_explanation bucket.count search.count search.time search.bucket_avg   search.exp_avg_hour node.id node.name node.ephemeral_id node.address
datafeed-low_request_rate    stopped                        1457         3            62ms        0.0425531914893617  20.160978098329824                                      
datafeed-response_code_rates stopped                        1460         18           127ms       0.08698630136986302 2.529951653336354                                       
datafeed-url_scanning        stopped                        1460         18           173ms       0.1184931506849315  2.8745993097731013                                      

# GET _cat/ml/datafeeds?help
id                     |                       | the datafeed_id                                           
state                  | s                     | the datafeed state                                        
assignment_explanation | ae                    | why the datafeed is or is not assigned to a node          
bucket.count           | bc,bucketCount        | bucket count                                              
search.count           | sc,searchCount        | number of searches ran by the datafeed                    
search.time            | st,searchTime         | the total search time                                     
search.bucket_avg      | sba,bucketTimeMin     | the average search time per bucket (millisecond)          
search.exp_avg_hour    | seah,searchExpAvgHour | the exponential average search time per hour (millisecond)
node.id                | ni,nodeId             | id of the assigned node                                   
node.name              | nn,nodeName           | name of the assigned node                                 
node.ephemeral_id      | ne,nodeEphemeralId    | ephemeral id of the assigned node                         
node.address           | na,nodeAddress        | network address of the assigned node                      


```

closes https://github.com/elastic/elasticsearch/issues/51411

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/cat/RestCatDatafeedsAction.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        return "cat_ml_get_datafeeds_action";
+    }
+
+    @Override
+    protected RestChannelConsumer doCatRequest(RestRequest restRequest, NodeClient client) {
+        String datafeedId = restRequest.param(DatafeedConfig.ID.getPreferredName());
+        if (Strings.isNullOrEmpty(datafeedId)) {
+            datafeedId = GetDatafeedsStatsAction.ALL;
+        }
+        GetDatafeedsStatsAction.Request request = new GetDatafeedsStatsAction.Request(datafeedId);
```
Comment (davidkyle) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/rest/cat/RestCatDatafeedsAction.java]: I thought this was `GetDatafeeds` not stats but stats makes a lot more sense 👍. I suppose naming the endpoint `_cat/ml/datafeeds_stats` isn't great. All good here.

File: x-pack/plugin/src/test/resources/rest-api-spec/api/cat.ml.datafeeds.json [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        }
+      ]
+    },
+    "params":{
+      "allow_no_datafeeds":{
+        "type":"boolean",
+        "required":false,
+        "description":"Whether to ignore if a wildcard expression matches no datafeeds. (This includes `_all` string or when no datafeeds have been specified)"
+      },
+      "bytes":{
```
Comment (davidkyle) [inline x-pack/plugin/src/test/resources/rest-api-spec/api/cat.ml.datafeeds.json]: Are there any byte units returned?

Comment (benwtrent) [reply x-pack/plugin/src/test/resources/rest-api-spec/api/cat.ml.datafeeds.json]: Don't think so, these options are a catch all as these parameters come by default with _cat APIs.

Comment (davidkyle) [reply x-pack/plugin/src/test/resources/rest-api-spec/api/cat.ml.datafeeds.json]: There is also a default `size` parameter https://www.elastic.co/guide/en/elasticsearch/reference/current/cat.html that is only used by [cat thread pool](https://github.com/elastic/elasticsearch/blob/91663116221052b9aa05cc183d80c1e546073371/rest-api-spec/src/main/resources/rest-api-spec/api/cat.thread_pool.json#L35)

My preference would be to remove this but it's not a deal breaker

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Comment by lcawl** [comment]
Documentation added in https://github.com/elastic/elasticsearch/pull/52738

**Pull request merged**